pkg/controller/issueapi/validate_code.go

```diff
@@ -109,7 +109,7 @@ func (c *Controller) BuildVerificationCode(ctx context.Context, internalRequest
 	var smsProvider sms.Provider
 	if !request.OnlyGenerateSMS && request.Phone != "" {
 		var err error
-		smsProvider, err = realm.SMSProvider(c.db)
+		smsProvider, err = realm.SMSProvider(c.db, vCode.IsUserReport())
```

Comment:
I don't think we care _which_ sms provider we get back here, just that we got one.